hudi-common/src/main/java/org/apache/hudi/common/table/read/BufferedRecordMergerFactory.java

```diff
@@ -113,16 +113,11 @@ public Option<DeleteRecord> deltaMerge(DeleteRecord deleteRecord, BufferedRecord
 
     @Override
     public Pair<Boolean, T> finalMerge(BufferedRecord<T> olderRecord, BufferedRecord<T> newerRecord) {
-      if (newerRecord.isCommitTimeOrderingDelete()) {
-        return Pair.of(true, newerRecord.getRecord());
-      }
-      Comparable newOrderingValue = newerRecord.getOrderingValue();
-      Comparable oldOrderingValue = olderRecord.getOrderingValue();
-      if (!olderRecord.isCommitTimeOrderingDelete()
-          && oldOrderingValue.compareTo(newOrderingValue) > 0) {
+      if (shouldKeepNewerRecord(olderRecord, newerRecord)) {
+        return Pair.of(newerRecord.isDelete(), newerRecord.getRecord());
+      } else {
```

Comment:
nit: could directly `return Pair.of(olderRecord.isDelete(), olderRecord.getRecord());` outside if branch for brevity.